### PR TITLE
Update to Gradle 8.0

### PR DESCRIPTION
**Pull request by romtsn**
_#skip-changelog_

Short summary:
* Most of the root project's `build.gradle.kts` changes are to fix the implicit dependency between tasks problem (e.g. we're not publishing `-debug` artifacts, spotless kotlin should not run over the `sentry-native` submodule, etc. etc.)
* Most of the dependency version updates is to support Kotlin 1.8.0, which Gradle 8 ships with
* jacoco configuration changes are introduced by Gradle 8
* Removed duplicated KotlinCompile tasks config for some backend modules
* Replaced the spring boot BOM plugin with `platform`, as the BOM plugin wasn't compatible with Gradle 8, but `platform` is technically the same BOM but without the plugin, so that just works

### CODE REVIEW

**Comment by romtsn** [comment]
The PR is blocked by spotless https://github.com/diffplug/spotless/issues/1572 due to an implicit dependency between spotlessKotlinGradle and buildCMake tasks. I'm thinking of temporarily disabling the formatting `.gradle.kts` files for now - this should work the issue around. Or we can wait on spotless for a couple of more weeks to see if they'll fix the issue or provide a different workaround.

**Review by markushi — APPROVED** [review]
LGTM!

**Comment by adinauer** [comment]
Converted to draft as this breaks samples in IDE with error message:

```
Toolchain from `executable` property does not match toolchain from `javaLauncher` property
```

It seems using a preview version of Android Studio it's possible to select the embedded JDK of the IDE to make it work. In Android Studio Electric Eel | 2022.1.1 Patch 2 however it doesn't work as the embedded JDK is too old.

There's this https://github.com/gradle/gradle/issues/22845 and https://stackoverflow.com/questions/75715341/toolchain-from-executable-property-does-not-match-toolchain-from-javalauncher where people are experiencing the same / similar problem.

**Comment by romtsn** [comment]
ok @adinauer I've made it work, check https://github.com/getsentry/sentry-java/pull/2563/commits/4e24af9cfb969f10e9bd7898b6ad6c1e0ae74e1c. Dunno if it breaks something, but I didn't notice anything wrong/weird, since it should only affect these specific IJ run configurations. But you better test it as well, thanks.

**Review by adinauer — APPROVED** [review]
Seems to be working now. Great job.

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,11 @@
 - Fix timestamps of PerformanceCollectionData in profiles ([#2632](https://github.com/getsentry/sentry-java/pull/2632))
 - Fix missing propagateMinConstraints flag for SentryTraced ([#2637](https://github.com/getsentry/sentry-java/pull/2637))
 
+### Dependencies
+- Bump Gradle from v7.6.0 to v8.0.2 ([#2563](https://github.com/getsentry/sentry-java/pull/2563))
+    - [changelog](https://github.com/gradle/gradle/blob/master/CHANGELOG.md#v802)
+    - [diff](https://github.com/gradle/gradle/compare/v7.6.0...v8.0.2)
```
Comment (romtsn) [inline CHANGELOG.md:19]:
```suggestion
### Dependencies
- Bump compile Kotlin version to 1.8.0
- Bump compile Compose version to 1.3.0
- Bump Gradle from v7.6.0 to v8.0.2 ([#2563](https://github.com/getsentry/sentry-java/pull/2563))
    - [changelog](https://github.com/gradle/gradle/blob/master/CHANGELOG.md#v802)
    - [diff](https://github.com/gradle/gradle/compare/v7.6.0...v8.0.2)
```

**Pull request merged**